src/ui/map.js

```diff
@@ -580,6 +580,32 @@ class Map extends Camera {
         return this.transform.pointLocation(Point.convert(point));
     }
 
+    /**
+     * Returns true if the map is panning, zooming, rotating, or pitching due to a camera animation or user gesture.
+     */
+    isMoving(): boolean {
+        return this._moving ||
+            this.dragPan.isActive() ||
+            this.dragRotate.isActive() ||
+            this.scrollZoom.isActive();
+    }
+
+    /**
+     * Returns true if the map is zooming due to a camera animation or user gesture.
+     */
+    isZooming(): boolean {
+        return this._zooming ||
+            this.scrollZoom.isActive();
```

Comment:
should `this.doubleClickZoom.isActive()` also be called here?

Reply:
`DoubleClickZoomHandler` implements the zoom using `zoomTo`, which will set `_zooming`, so it isn't strictly necessary. (In fact I think `DoubleClickZoomHandler` should probably not have an `isActive()` method at all. Unlike drag panning/zooming, where there's an interval bookended by down/up events, double-clicking is an instantaneous action.)

Reply:
`DoubleClickZoomHandler #isActive()` turns out to be used internally, so I'll abandon that idea for now.